crosscorrelation.Rmd

```diff
@@ -0,0 +1,384 @@
+---
+title: "uploadid"
+author: "Amenze"
+date: "June 28, 2018"
+output: html_document
+---
+
+```{r}
+setwd("/Users/Amenze/Desktop/tidepool/refdata")
+```
+
+
+```{r }
+#library(data.table)
+library(ggplot2)
+library(plyr)
+library(dplyr)
+#using jmotif
+library(jmotif)
+#install.packages("RecordLinkage")
+library(RecordLinkage)
+#install.packages("PTXQC")
+library("PTXQC")
+library(stringr)
+library(zoo)
+
+
+
+```
+
+
+
+
+```{r}
+
+# create a list from these files
+list.filenames<-list.files("/Users/Amenze/Desktop/tidepool/refdata",pattern=".csv$")
+
+
+#extract files based on duplicated utctime
+for (i in 1:length(list.filenames))
+{
+  patient<-read.csv(list.filenames[i])
+  patient_cbg<-subset(patient,patient$type=="cbg")
+  if ((length(unique(patient_cbg$utcTime)))!=nrow(patient_cbg))
+      write.csv(patient_cbg,paste0("/Users/Amenze/Desktop/tidepool/refdata/duplicated",list.filenames[i]))
+
+ }
+```
+
+
+
+
+
+```{r}
+##read in file and check for duplicated utc
+
+field<-c("deviceId","id","uploadId","utcTime","type","value")
+patient<-read.csv("duplicated0289cfb8bd6d61ccf1f31c07aa146b7b14f0eb74474be4311860d9d77dd30f15.csv")[,field]
+patient$mgvalue<-patient$value*18.01559
+length(unique(patient$uploadId)) #1319
+length(unique(patient$id))==nrow(patient) ##True
+length(unique(patient$utcTime))#72287
+length(unique(patient$utcTime))==nrow(patient) #False
+
+
+
+###subset patient file based on  5 uploadids
+df1<-subset(patient,patient$uploadId=="upid_3c41703c2d3a8b97f479afdb6ccf799f")
+
+df2<-subset(patient,patient$uploadId=="upid_3fc32e5ad912a8ea7efced9151804bdb")
+
+
+df3<-subset(patient,patient$uploadId=="upid_17db2d2a0ae0e02a12c0a5067e5fe85b")
+
+
+df4<-subset(patient,patient$uploadId=="upid_5fad608cf32bd03a1cd56e3bb1fdb834")
+
+
+
+df5<-subset(patient,patient$uploadId=="upid_830c6de3e2ecbbec6fbad0cecc64bdf5")
+
+#plot values for each uploadid
+par(mfrow=c(3,3))
+
+
+plot.ts(df1$value)
+plot.ts(df2$value)
+plot.ts(df3$value)
+plot.ts(df4$value)
+plot.ts(df5$value)
+
+```
+
+
+
+```{r}
```

Comment:
Let's move these key functions out to an `.R` file that can be included in this Rmarkdown.  This way, we can eventually clean up the functions more and make them a library that someone can import.